File: Python/Pesel.py
t = int(input())
weights = [1, 3, 7, 9, 1, 3, 7, 9, 1, 3, 1]
for _ in range(t):
    pesel = input().strip()
    result = sum(int(pesel[i]) * weights[i] for i in range(11))
    if result % 10 == 0:
        print('D')
    else:
        print('N')

File: Java/Pesel.java
import java.util.Scanner;

class Main
{
    public static void main(String[] args)
    {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        input.nextLine();
        int[] weights = {1, 3, 7, 9, 1, 3, 7, 9, 1, 3, 1};
        for (int i = 0; i < t; i++) 
        {
            String pesel = input.nextLine().trim();
            int result = 0;
            for (int j = 0; j < 11; j++)
                result += Character.getNumericValue(pesel.charAt(j)) * weights[j];
            if (result % 10 == 0)
                System.out.println("D");
            else
                System.out.println("N");
        }

        input.close();
    }
}